server/api/unsafe_operation.go

```diff
@@ -43,15 +44,43 @@ func newUnsafeOperationHandler(svr *server.Server, rd *render.Render) *unsafeOpe
 // @Router /admin/unsafe/remove-failed-stores [POST]
 func (h *unsafeOperationHandler) RemoveFailedStores(w http.ResponseWriter, r *http.Request) {
 	rc := getCluster(r)
-	var stores map[uint64]string
-	if err := apiutil.ReadJSONRespondError(h.rd, w, r.Body, &stores); err != nil {
-		return
+	var input map[string]interface{}
+	if err := apiutil.ReadJSONRespondError(h.rd, w, r.Body, &input); err != nil {
+		h.rd.JSON(w, http.StatusBadRequest, "Cannot parse the request")
+		return;
 	}
-	if len(stores) == 0 {
+	rawStores, exists := input["stores"]
+	if !exists {
 		h.rd.JSON(w, http.StatusBadRequest, "No store specified")
 		return
 	}
-	if err := rc.GetUnsafeRecoveryController().RemoveFailedStores(stores); err != nil {
+	storeList, ok := rawStores.([]string)
+	if !ok || len(storeList) == 0 {
+		h.rd.JSON(w, http.StatusBadRequest, "Invalid input for the stores")
+		return
+	}
+	stores := make(map[uint64]string)
+	for _, storeStr := range storeList {
+		storeID, err := strconv.ParseUint(storeStr, 10, 64)
+		if err != nil {
+			h.rd.JSON(w, http.StatusBadRequest, "Cannot parse the store ID")
+			return
+		}
+		stores[storeID] = ""
+	}
+	var pauseDuration int64
+	rawPauseDuration, exists := input["pause"]
```

Comment:
let's call it `timeout`, also indicate the max execution duration of recovery

Reply:
Done.